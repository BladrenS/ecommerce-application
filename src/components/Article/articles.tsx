import { blog1alt, blog2 } from '../../assets';

export const articleItems: { header: string; image: string; text: string }[] = [
  {
    header: '🏍 Yamaha R1 2025: Don’t Miss the Updated Model of the Iconic R1 Series!',
    image: blog1alt,
    text: `The Yamaha YZF-R1 has always stood as a symbol of cutting-edge performance and aggressive design — and the 2025 model raises the bar even higher. Whether you're a seasoned rider, a track enthusiast, or simply someone who admires legendary superbikes, the R1 2025 is not something you want to miss.

🔧 What’s New in 2025? 

Yamaha has refined the R1 with subtle yet impactful updates:

- Enhanced aerodynamics with revised fairing lines for improved wind resistance and stability at high speeds.
- Optimized electronics package including refined ride modes, updated traction control, and cornering ABS for razor-sharp precision.
- A newly tuned crossplane engine delivers smoother throttle response and more linear power through the rev range.
- New colorways and graphics give the R1 a fresh, futuristic look without losing its aggressive stance.

🏁 Track-Born, Road-Ready

Born on the MotoGP grid and bred for the streets, the 2025 R1 continues to bridge the gap between factory racing machines and everyday performance bikes. Its chassis remains razor-sharp, while the suspension system has been further refined for better feedback and control — whether you're carving corners on a canyon road or pushing limits on a closed circuit.

👀 Why You Should Pay Attention

With ever-tougher competition from rivals like the CBR1000RR-R Fireblade, ZX-10R, and Panigale V4, Yamaha’s commitment to innovation ensures the R1 stays in the fight — and often ahead. For fans of the brand and performance junkies alike, the 2025 Yamaha R1 is a worthy evolution of a true icon.

🧠 Technology Highlights

- 6-axis IMU for superior traction and stability control in any condition.
- Full-color TFT display with customizable layout and telemetry options.
- Lightweight aluminum Deltabox frame for perfect balance between rigidity and flexibility.
- Optional quick shifter and auto-blipper for lightning-fast gear changes.

🎨 Design & Feel

The 2025 R1 embraces a more futuristic design with sharper contours, aggressive front fascia, and LED matrix headlights. Every line has a purpose — from channeling airflow to reducing drag. It's not just a motorcycle — it’s an extension of your style and personality.

⚡ Ready to ride the future?

Whether you're a collector, a rider hungry for the track, or someone seeking the ultimate supersport experience — this bike belongs on your radar. Pre-orders are open now in our market. Don't just dream it. Ride it.,

`,
  },
  {
    header: 'Kawasaki ninja H2R: The Untamed Hyperbike Redefining Speed',
    image: blog2,
    text: `The Kawasaki Ninja H2R isn’t just a motorcycle — it’s a force of nature. Built for the track and unconstrained by road regulations, the H2R is the most extreme expression of Kawasaki’s engineering, performance, and aerodynamic mastery.

⚙️ Unmatched Power and Engineering

The H2R is powered by a **supercharged 998cc inline-four engine**, producing a jaw-dropping **over 300 horsepower**. Yes, you read that right. This is not your average superbike — this is a hyperbike engineered to push the limits of physics.

- Supercharged engine developed in-house by Kawasaki Heavy Industries.
- Aerospace-grade components including carbon-fiber wings and a trellis frame designed for insane rigidity.
- Track-only focus with no compromises for comfort — just raw speed and precision.

✈️ Aerodynamics Inspired by Aviation

The H2R's design borrows from **aerospace technology**, developed by Kawasaki's aerospace division:

- **Carbon-fiber winglets** on the fairings provide high-speed downforce.
- Sharp, angular lines reduce drag and enhance airflow.
- The trellis frame exposes just enough of its rawness to hint at what’s beneath.

🧠 Advanced Electronics & Rider Aids

Even with its sheer power, Kawasaki ensures control through a suite of cutting-edge electronics:

- Kawasaki Cornering Management Function (KCMF)
- Launch Control Mode (KLCM)
- Intelligent ABS and Engine Brake Control
- Quick shifter for seamless up and down gear changes

🔥 Track-Only, Fearlessly Free

The H2R is **not street-legal** — and that’s exactly the point. This machine is a statement: for riders who want nothing but the purest form of speed, sound, and adrenaline. It’s built to dominate racetracks and leave everything else behind in a cloud of turbocharged fury.

🏆 A Legend Among Legends

With few peers in the two-wheeled world, the Ninja H2R has etched its name into motorcycle history. Whether you’re watching it scream down a runway or lucky enough to ride one, the experience is unforgettable.

💬 Final Thoughts

The Kawasaki Ninja H2R isn’t for everyone — and that’s what makes it so special. It’s for the fearless. The focused. The few who crave the extremes of performance and engineering.

Are you ready to tame the wildest Ninja ever built?`,
  },
];
