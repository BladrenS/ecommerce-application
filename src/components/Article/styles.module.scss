.container {
  display: flex;
  flex-direction: column;
  align-items: center;

  width: 95%;
  margin: 0 auto;
  padding: 150px 0;
}

.header {
  margin-bottom: 30px;
  color: var(--primary-color);
}

.image {
  width: 100%;
}

.text {
  width: 70%;
  margin-top: 30px;

  line-height: 1.6;
  color: var(--secondary-color);
  white-space: pre-wrap;
}
