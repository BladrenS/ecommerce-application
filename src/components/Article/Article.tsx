import { useParams } from 'react-router-dom';

import { articleItems } from './articles';
import styles from './styles.module.scss';

export const Article = () => {
  const { id } = useParams();
  const index = Number(id);
  const article = articleItems[index];

  return (
    <div className={styles.container}>
      <h1 className={styles.header}>{article.header}</h1>
      <img src={article.image} alt="article" className={styles.image} />
      <div className={styles.text}>{article.text}</div>
    </div>
  );
};
