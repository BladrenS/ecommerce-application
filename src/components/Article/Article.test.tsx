import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

import { Article } from './Article';

jest.mock('./articles', () => ({
  articleItems: [
    {
      header: 'Test Article 0',
      image: 'image0.jpg',
      text: 'Content for article zero',
    },
    {
      header: 'Test Article 1',
      image: 'image1.jpg',
      text: 'Content for article one',
    },
  ],
}));

describe('Article snapshot test', () => {
  it('renders article with id 0 correctly', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/article/0']}>
        <Routes>
          <Route path="/article/:id" element={<Article />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders article with id 1 correctly', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/article/1']}>
        <Routes>
          <Route path="/article/:id" element={<Article />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
