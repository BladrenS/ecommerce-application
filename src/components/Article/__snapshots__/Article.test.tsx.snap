// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article snapshot test renders article with id 0 correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      class="header"
    >
      Test Article 0
    </h1>
    <img
      alt="article"
      class="image"
      src="image0.jpg"
    />
    <div
      class="text"
    >
      Content for article zero
    </div>
  </div>
</DocumentFragment>
`;

exports[`Article snapshot test renders article with id 1 correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      class="header"
    >
      Test Article 1
    </h1>
    <img
      alt="article"
      class="image"
      src="image1.jpg"
    />
    <div
      class="text"
    >
      Content for article one
    </div>
  </div>
</DocumentFragment>
`;
